test/provider/AppDirectory.unittest.ts

```diff
@@ -0,0 +1,109 @@
+import 'jest';
+import 'reflect-metadata';
+
+import {AppDirectory} from '../../src/provider/model/AppDirectory';
+import {Application} from '../../src/client/directory';
+
+
+enum StorageKeys {
+    URL = 'url',
+    APPLICATIONS = 'applications'
+}
+
+const appTemplate: Application = {
+    appId: 'id',
+    name: 'test-app',
+    manifest: '',
+    manifestType: '',
+    intents: [
+        {
+            name: 'Call',
+            contexts: [
+                'dial'
+            ],
+            customConfig: {}
+        }
+    ]
+};
+
+// Generate applications
+const generateApplications = (amount: number): Application[] => {
+    return Array(amount).fill(appTemplate).map((app, i) => ({
+        ...app,
+        appId: `id-${i}`,
+        name: `${app.name}-${i}`
+    }));
+};
+
+describe('AppDirectory Unit Tests', () => {
+    let appDirectory: AppDirectory;
+    let store: LocalForage;
+
+    afterEach(async () => {
+        await store.clear();
+    });
+
+    describe('Storage', () => {
+        beforeEach(async () => {
+            appDirectory = new AppDirectory();
+        });
+
+        afterEach(async () => {
+            await store.clear();
+        });
+
+        it('Correctly sets up the store', async () => {
```

Comment:
Can you set up the tests so that the `describe`s and `it`s read as a (fairly concise) sentence? The describes should describe the context, and the it's the expected state/behaviour of the app when it's in that state.

For example, this test has the "full" name: `AppDirectory Unit Tests > Storage > Correctly sets up the store`. Whilst keeping the exact same code (only changing the describe/it strings), this could be more descriptively titled `AppDirectory > When directory is first initialised > Nothing exists within storage`.

Naming in this way also points out some edge cases - this test could fail if I run it immediately after running the demo app, since the storage would persist from the demo app into the test environment. There's also no tests in here that assume there is something in storage when the service starts.